File: app/schemas/audit.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AuditResponse(BaseModel):
    id: int
    user_id: int | None
    action: str
    entity_type: str
    entity_id: int | None
    description: str
    old_value: dict | None = None
    new_value: dict | None = None
    ip_address: str | None = None
    request_method: str | None = None
    endpoint: str | None = None
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)


class VersionResponse(BaseModel):
    id: int
    decision_id: int
    version_number: int
    title: str
    problem_statement: str
    category: str
    status: str
    rationale: str | None = None
    created_by: int
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/services/audit.ts
import api from "./api";

export interface AuditLog {
  id: number;
  user_id?: number | null;
  action: string;
  entity_type: string;
  entity_id?: number | null;
  description?: string | null;
  ip_address?: string | null;
  created_at: string;
  old_value?: string | null;
  new_value?: string | null;
  request_method?: string | null;
  endpoint?: string | null;
}

export interface AuditLogResponse {
  items: AuditLog[];
  page: number;
  page_size: number;
  total: number;
}

export interface AuditFilters {
  user_id?: number;
  action?: string;
  entity_type?: string;
  entity_id?: number;
  start_date?: string;
  end_date?: string;
  page?: number;
  page_size?: number;
}

export async function getAuditLogs(
  filters?: AuditFilters,
) {
  const response = await api.get("/audit-logs", {
    params: filters,
  });

  return response.data as AuditLogResponse;
}